api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -75,6 +77,19 @@ public interface OverwriteFiles extends SnapshotUpdate<OverwriteFiles> {
    */
   OverwriteFiles deleteFile(DataFile file);
 
+  /**
+   * Deletes a set of data files from the table with their respective delete files.
+   *
+   * @param dataFilesToDelete the data files to be deleted from the table
+   * @param deleteFilesToDelete the delete files corresponding to the data files to be deleted from
+   *     the table
+   * @return this for method chaining
+   */
+  default OverwriteFiles deleteFiles(
```

Comment:
let's call this `removeFiles`.

See PR #13814 for the context

Reply:
#13184 is mainly for RowDelta, while naming this `deleteFiles()` matches the other method `deleteFile()` in this and in other APIs (such as `RewriteFiles`)

Reply:
that is fair to be consistent with the existing method. 

But I still feel `removeFile` is a better name than `deleteFile` as deleteFile can be associated with just the position/equality delete file.